### ISSUE TITLE
[Bug] Some devices can’t see each other—Windows 11 24H2 involved?

### ISSUE THREAD

**Issue by danyonline64** [schlagmichdoch/PairDrop#352]
Some devices are unable to detect each other on the network. This issue started recently and affects specific machines, including one running Windows 11 24H2.

    Open Pairdrop on device A (e.g., Windows 11 24H2).
    Open Pairdrop on device B (another device on the same network).
    Attempt to detect device B from device A (or vice versa).
    Devices fail to see each other.

All devices on the same network should detect each other and allow file sharing via Pairdrop, as they did before.

    OS: Windows 11 24H2
    Browser: Chrome/Firefox
    Version: last

Bug occurs on official PairDrop instance https://pairdrop.net/
No

Bug occurs on self-hosted PairDrop instance
Yes

Self-Hosted Setup

    Proxy: Nginx
    Deployment: Docker Compose
    Version: v1.10.10

Additional context
The issue seems to be new and might be related to changes in the Windows 11 24H2 update. Other devices on the same network (using the same Pairdrop instance) work fine. Could this be related to networking or compatibility issues with this OS version?

Thanks :)

**Comment by schlagmichdoch** [comment]
As PairDrop is a web app it should not be OS but browser specific but there might be some firewall or vpn settings that affect visibility.

What do your browser logs on the affected device say? Does it try to connect but ultimately fail?

Have you deployed your own TURN server via `RTC_CONFIG` env var or do you use the default settings?

**Comment by JonSnow88** [comment]
I have the same issue but both devices are running 23H2, weirdly both devices can see my phone. So was able to transfer from Windows > iOS > Windows.  

It's just these two devices that cannot see each other, the rest are fine.

**Comment by JonSnow88** [comment]
This is self-hosted, have not tried pairdrop.net.

No turn server installed, but one of the Windows hosts does have a VPN installed. Although this does not explain why the VPN Windows host can see my Phone, but the two Windows host cannot see each other. My phone can talk to all Windows host without issue.

The VPN is setup to pass local traffic, RDP, SMB etc. all work with the VPN enabled.

**Comment by schlagmichdoch** [comment]
Just try https://pairdrop.net on both devices. If it works there but not on your self hosted instance you will need a TURN server.

Also, what do the browser console logs say when you try to connect those two windows machines?

**Comment by marcusdoc** [comment]
**tldr regarding my entry (this fixes the problem)**
Firefox: about:config
media.peerconnection.ice.obfuscate_host_addresses = false

**now the AI's report**
## Firefox mDNS obfuscation causes ICE failure on LAN — devices invisible

### Environment
- **PairDrop:** v1.11.2 (self-hosted, behind Nginx Proxy Manager with TLS)
- **OS:** Windows 11 25H2
- **Browser:** Firefox 148.0.2
- **Network:** Mixed 10GbE / 1GbE, single subnet, no VLANs

### Problem
Firefox users on the same LAN cannot see other devices in PairDrop. The WebSocket connection succeeds, peers are announced (`type: "peers"` received with correct `roomId: "127.0.0.1"`), and SDP/ICE signaling is exchanged — but the WebRTC data channel never opens, so peers never appear in the UI.

### Root Cause
Firefox's `media.peerconnection.ice.obfuscate_host_addresses` (enabled by default) replaces local IP candidates with mDNS `.local` hostnames. When the remote peer cannot resolve these mDNS hostnames (common in cross-browser scenarios on LAN), ICE negotiation fails silently.

**`about:webrtc` showed:**
- ICE State: **failed**
- Local Candidate: `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.local:55503/udp(host) [non-proxied]`
- Remote Candidate: `<public STUN IP>:53855/udp(srflx)`
- Bytes sent/received: **0 / 0**

### Workaround
In Firefox `about:config`, set:
```
media.peerconnection.ice.obfuscate_host_addresses = false
```
Devices appear immediately after reconnecting.

### Notes
- Chromium-based browsers (Edge) on the same machine work without issue
- This is related to Mozilla Bug [[#1698141](https://bugzilla.mozilla.org/show_bug.cgi?id=1698141)](https://bugzilla.mozilla.org/show_bug.cgi?id=1698141) — mDNS ICE candidates breaking WebRTC P2P on private LANs
- Affects self-hosted instances; may also affect pairdrop.net
- Likely connected to issue #352 (Windows 11 24H2 discovery failures — the reporter was also using Firefox)

### Suggestion
Consider adding a note to the FAQ/troubleshooting docs about this Firefox default, or detecting the ICE failure and surfacing a user-facing hint.
